cmd/storage-consumer/main.go

```diff
@@ -334,86 +344,72 @@ func (c *consumer) difference(map1, map2 map[dmlPathKey]uint64) map[dmlPathKey]f
 	return resMap
 }
 
-// getNewFiles returns dml files in specific ranges.
+// getNewFiles returns newly created dml files in specific ranges
 func (c *consumer) getNewFiles(ctx context.Context) (map[dmlPathKey]fileIndexRange, error) {
-	m := make(map[dmlPathKey]fileIndexRange)
+	tableDMLMap := make(map[dmlPathKey]fileIndexRange)
 	opt := &storage.WalkOption{SubDir: ""}
 
-	origTableMap := make(map[dmlPathKey]uint64, len(c.tableIdxMap))
-	for k, v := range c.tableIdxMap {
-		origTableMap[k] = v
+	origDMLIdxMap := make(map[dmlPathKey]uint64, len(c.tableDMLIdxMap))
+	for k, v := range c.tableDMLIdxMap {
+		origDMLIdxMap[k] = v
 	}
 
-	schemaSet := make(map[schemaPathKey]struct{})
 	err := c.externalStorage.WalkDir(ctx, opt, func(path string, size int64) error {
 		var dmlkey dmlPathKey
 		var schemaKey schemaPathKey
+		var fileIdx uint64
+		var err error
 
 		if strings.HasSuffix(path, "metadata") {
 			return nil
 		}
 
 		if strings.HasSuffix(path, "schema.json") {
-			err := schemaKey.parseSchemaFilePath(path)
+			err = schemaKey.parseSchemaFilePath(path)
 			if err != nil {
 				log.Error("failed to parse schema file path", zap.Error(err))
-			} else {
-				schemaSet[schemaKey] = struct{}{}
+				// skip handling this file
+				return nil
+			}
+			// fake a dml key for schema.json file
+			dmlkey.schemaPathKey = schemaKey
+			dmlkey.partitionNum = fakePartitionNumForSchemaFile
```

Comment:
Please add a comment to tell why we need to set it `fakePartitionNumForSchemaFile `.

Reply:
done